src/main/java/com/xceptance/neodymium/common/browser/BrowserBeforeRunner.java

```diff
@@ -0,0 +1,131 @@
+        {
+            browserRunner.setUpTest();
+
+        }
+        else if (startNewBrowserForSetUp)
+        {
+            throw new RuntimeException("No browser setting for " + (junit5 ? "@BeforeEach" : "@Before") + " method '" + (before.getName()) + "' was found."
+                                       + " If browser is suppressed for the test"
+                                       + " but the before method is annotated with @StartNewBrowserForSetUp"
+                                       + " because it requires a browser, please,"
+                                       + " use @Browser annotation to specify what browser is required for this method.");
+        }
+        boolean beforeFailed = false;
+        try
+        {
+            Throwable e = beforeMethodInvocation.get();
+            if (e != null)
+            {
+                beforeFailed = true;
+                throw e;
+            }
+        }
+        finally
+        {
+            // if we did a set up of new driver before the @Before method, we need to close it
+            if (startNewBrowserForSetUp && browserConfiguration != null)
+            {
+                browserRunner.teardown(beforeFailed);
+            }
+
+            // set driver back to the original to execute the test or clean up (in case of failure)
+            // Neodymium.setWebDriverStateContainer(oldWDsCont);
+            if (oldWDsCont != null)
+            {
+                WebDriverRunner.setWebDriver(oldWDsCont.getWebDriver());
+                Neodymium.setWebDriverStateContainer(oldWDsCont);
+            }
+            if (oldBrowserConfiguration != null)
+            {
+                Neodymium.setBrowserProfileName(oldBrowserConfiguration.getConfigTag());
+                Neodymium.setBrowserName(oldBrowserConfiguration.getCapabilities().getBrowserName());
+            }
+        }
+    }
+
+    public static boolean shouldStartNewBrowser(Method each)
+    {
+        List<StartNewBrowserForSetUp> methodStartNewBrowserForSetUp = Data.getAnnotations(each,
+                                                                                          StartNewBrowserForSetUp.class);
+        List<StartNewBrowserForSetUp> classStartNewBrowserForSetUp = Data.getAnnotations(each.getDeclaringClass(),
+                                                                                         StartNewBrowserForSetUp.class);
+
+        // if global config for dontStartNewBrowserForSetUp is set to false, we should not reach this point
+        boolean startNewBrowserForSetUp = false;
+        if (!classStartNewBrowserForSetUp.isEmpty())
+        {
+            startNewBrowserForSetUp = true;
+        }
+
+        if (!methodStartNewBrowserForSetUp.isEmpty())
+        {
+            startNewBrowserForSetUp = true;
+        }
+
+        // if @Before method is annotated with @SuppressBrowser annotation, no new browser should be started
```

Comment:
again the code does not do what the comment demands, does it? there is no check for suppressBrowser here